### ISSUE TITLE
Fastify types throwing compile error

### ISSUE THREAD

**Issue by ppetitto1** [fastify/session#94]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.29.0

### Plugin version

8.2.0

### Node.js version

16.x

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

11.3.1

### Description

```
node_modules/@fastify/session/types/types.d.ts(9,22): 
error TS2694: 
Namespace '"/Users/Projects/gcp-server/node_modules/fastify/fastify"' has no exported member 'Session'.
```

Seeing this type error. 

### Steps to Reproduce

run tsc and compile 

### Expected Behavior

compiled successfully

**Comment by mcollina** [comment]
Can you provide steps to reproduce? We often need a reproducible example, e.g. some code that allows someone else to recreate your problem by just copying and pasting it. If it involves more than a couple of different file, create a new repository on GitHub and add a link to that.

**Comment by mrbrianevans** [comment]
I had the same issue, and after a bit of fiddling around I managed to solve it.

The problem was in my type declaration files for Session and possibly any other fastify modules I was using special types for. In my case this was `session.d.ts` but could be named anything. Containing this:
```typescript
declare module "fastify" {
  interface Session {
    name: string
  }
}
```
but was fixed by 2 things:
 1. add `export {}` to your type declaration file (to make it a module or something)
 2. combine all the `declare module 'fastify'` blocks in your application.

I was using `@fastify/oauth2` and I had two seperate `declare module 'fastify'` blocks (one for session and one for oauth), and it was causing issues, but when I combined them and added an export to the file, the build started working again. 

Also, using `request.session.get('key')` did not work with typescript for me. I had to rather use `request.session.key`.

Hope this helps @ppetitto1
